app/main.py

```diff
@@ -1,34 +1,68 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple, end: tuple, is_drowned: bool = False
+                 ) -> None:
+        self.decks = []
+        if start == end:
+            self.decks.extend([Deck(*start)])
+        elif start[1] != end[1]:
+            for deck in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], deck))
+        elif start[0] != end[0]:
+            for deck in range(start[0], end[0] + 1):
+                self.decks.append(Deck(deck, start[1]))
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def fire(self, row: int, column: int) -> (str, None):
+        result_get_deck = self.get_deck(row, column)
+        if result_get_deck is not None:
+            result_get_deck.is_alive = False
+            if not any([deck.is_alive for deck in self.decks]):
+                self.is_drowned = True
+                return "Sunk!"
+            return "Hit!"
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> (Deck, None):
```

Comment:
```suggestion
    def get_deck(self, row: int, column: int) -> Deck | None:
```

Reply:
not working 
And can you please explain me why in line 14 I cant leave like I did

Reply:
Than use Union from typing library